core/amber/src/main/python/core/models/schema/schema.py

```diff
@@ -0,0 +1,122 @@
+from typing import MutableMapping, Optional, Mapping, List, Tuple
+
+import pyarrow as pa
+
+from core.models.schema.attribute_type import (
+    AttributeType,
+    RAW_TYPE_MAPPING,
+    FROM_ARROW_MAPPING,
+    TO_ARROW_MAPPING,
+)
+
+
+class Schema:
+    """
+    Schema describes a sequence of attributes, maintaining a name-to-type mapping.
+
+    Schema is mapped to PyArrow's Schema (pyarrow.Schema), with each
+    AttributeType mapped to a pyarrow.DataType.
+
+    Schema is mapped to a Tuple Field (which is a collection TypeVar of
+    python objects).
+
+    See AttributeType for detailed mappings.
+
+    Note: Schema is to be used by the engine only, and should be invisible to the
+    users of Tuple. It only gets assigned to a Tuple to finalize the Tuple for
+    serialization purpose.
+    """
+
+    def __init__(
+        self,
+        arrow_schema: Optional[pa.Schema] = None,
+        raw_schema: Optional[Mapping[str, str]] = None,
+    ):
+        self._name_type_mapping: MutableMapping[str, AttributeType] = dict()
+
+        if arrow_schema is not None:
+            self.from_arrow_schema(arrow_schema)
+        if raw_schema is not None:
+            self._from_raw_schema(raw_schema)
```

Comment:
Minor comment: why is one named `from_arrow_schema` and the other `_from_raw_schema` (underscore prefix)?

Reply:
changed `from_arrow_schema` to private (i.e., adding the underscore prefix).